yt_idefix/data_structures.py

```diff
@@ -591,12 +591,14 @@ def _parse_definitions_header(self) -> None:
                 unit = unit_match.group(1).lower() + "_unit"
                 expr = unit_match.group(2)
                 # Before evaluating the expression, replace the input parameters,
-                # pre-defined constants, code units and sqrt function
+                # pre-defined constants, code units and arithmetic operators
                 # that cannot be resolved. The order doesn't matter.
                 expr = re.sub(r"g_inputParam\[(\w+)\]", self._get_input_parameter, expr)
                 expr = re.sub(r"CONST_\w+", self._get_constants, expr)
                 expr = re.sub(r"UNIT_(\w+)", self._get_unit, expr)
                 expr = re.sub(r"sqrt", "np.sqrt", expr)
+                expr = re.sub(r"log", "np.log", expr)
+                expr = re.sub(r"log10", "np.log10", expr)
```

Comment:
@xshaokun Can you also add a replacement of `pow` with `np.power` as well?
What is very common is that for example masses are put in say `pluto.ini` as 13.5 and in defeinitions then it is used as pow(10.,13.5)*CONST_Msun.
```suggestion
               expr = re.sub(r"log10", "np.log10", expr)
               expr = re.sub(r"pow", "np.power", expr)
```

Reply:
Ah because there's no problem to execute `pow(10, 13.5)` in Python, I think there's no need to add more code to replace it.

Reply:
> Also wanted to ask that will this not happen?
> 
> ```
> log10 -> np.log10
> ```
> 
> by line `600` and then
> 
> ```
> np.log10 -> np.np.log10
> ```
> 
> by line `601` creating problem Perhaps changing `log` to `np.log` will automatically account for `log10` as well.

Nice catch! I was really careless about this. Let's fix it right away. Thanks!

Reply:
@dutta-alankar Could you provide me a small patch of test data which includes log and log10 in unit definitions? I'll add them in test/data and you can also use them in the future.

Reply:
> @dutta-alankar Could you provide me a small patch of test data which includes log and log10 in unit definitions? I'll add them in test/data and you can also use them in the future.

Ah never mind. Currently we are only able to test vtk outputs. I'll try to modify the existing test data.